### PR TITLE
Add and test correct implementation for undefined const char values

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR

### CODE REVIEW

**Review by toregge — DISMISSED** [review]

File: searchlib/src/vespa/searchcommon/common/undefinedvalues.h [hunk 1]
```diff
@@ -31,43 +31,42 @@ inline constexpr const char* getUndefined<const char*>() {
...
-inline bool isUndefined<double>(const double & value) {
+inline bool isUndefined<double>(double value) {
     return std::isnan(value);
 }
 
-template <>
-inline bool isUndefined<vespalib::string>(const vespalib::string & value) {
-    return value.empty();
+inline bool isUndefined(const char * value) {
+    return (value == nullptr) || (value[0] == '\0');
```
Comment (toregge) [inline searchlib/src/vespa/searchcommon/common/undefinedvalues.h:69]: Note that value should never be nullptr here.

Comment (baldersheim) [reply searchlib/src/vespa/searchcommon/common/undefinedvalues.h:69]: So should probably just remove it  then...

Comment (toregge) [reply searchlib/src/vespa/searchcommon/common/undefinedvalues.h]: There are issues with multi value string attributes for documents with zero values
that might cause a crash if the check for nullptr is removed:

https://github.com/vespa-engine/vespa/blob/df7854e4c340ea814ca5cdf128fc31828a4b543b/searchlib/src/vespa/searchlib/attribute/multistringattribute.h#L62

this is also a problem for numeric multivalue attributes (wrong return values):

https://github.com/vespa-engine/vespa/blob/df7854e4c340ea814ca5cdf128fc31828a4b543b/searchlib/src/vespa/searchlib/attribute/multinumericattribute.h#L85
https://github.com/vespa-engine/vespa/blob/df7854e4c340ea814ca5cdf128fc31828a4b543b/searchlib/src/vespa/searchlib/attribute/multinumericattribute.h#L89
https://github.com/vespa-engine/vespa/blob/df7854e4c340ea814ca5cdf128fc31828a4b543b/searchlib/src/vespa/searchlib/attribute/multinumericattribute.h#L93
https://github.com/vespa-engine/vespa/blob/df7854e4c340ea814ca5cdf128fc31828a4b543b/searchlib/src/vespa/searchlib/attribute/multinumericenumattribute.h#L63

**Review by toregge — DISMISSED** [review]

File: searchlib/src/vespa/searchcommon/common/undefinedvalues.h (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchcommon/common/undefinedvalues.h]: Sigh..what have we not. The singlevalue methods should throw instead...

**Review by toregge — APPROVED** [review]

**Pull request merged**